include/pybind11/embed.h

```diff
@@ -262,14 +262,14 @@ class scoped_interpreter {
     }
 
 #if PY_VERSION_HEX >= 0x030B0000
-    explicit scoped_interpreter(const PyConfig& config,
-                                bool add_program_dir_to_path = true) {
+    explicit scoped_interpreter(const PyConfig &config, bool add_program_dir_to_path = true) {
         if (Py_IsInitialized() != 0) {
             pybind11_fail("The interpreter is already running");
         }
         PyStatus status = Py_InitializeFromConfig(&config);
         if (PyStatus_Exception(status)) {
-            throw std::runtime_error(PyStatus_IsError(status) ? status.err_msg : "Failed to init CPython");
+            throw std::runtime_error(PyStatus_IsError(status) ? status.err_msg
+                                                              : "Failed to init CPython");
         }
```

Comment:
This code appears to be entirely duplicated from the end on initialize. Furthermore, for consistency py::initialize should also be able to take a config on recent python versions. Therefore, I would suggest abstracting this into an overload for initialize(const PyConfig & config) that is only built when the python is recent enough. That would reduce code duplicate and make this a bit cleaner without changing the API that much.

Reply:
Thank you for your feedback. I find your suggestions reasonable.

However, I would rather get a sign of approval from the maintainers that they [want/do not mind] to have the feature introduced into the project before I rush into refactoring and extending tests. Is it at all possible to find out their opinion?

Reply:
Seem reasonable to me, @rwgk @henryiii thoughts?